build/scripts/version-update/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var releaseType string
+
+func init() {
+	flag.StringVar(&releaseType, "type", "", "Specify the release type ('before' or 'after')")
+}
+
+func main() {
+	flag.Parse()
+
+	if len(os.Args) < 3 {
+		log.Fatalf("Please provide the release type ('before' or 'after') and the initial version as command-line arguments")
+		return
+	}
+
+	releaseType := os.Args[1]
+	initialVersion := os.Args[2]
+
+	log.Printf("Release Type: %s", releaseType)
+	log.Printf("Initial Version: %s", initialVersion)
+
+	files := []string{
+		"install/helm/agones/Chart.yaml",
+		"install/yaml/install.yaml",
+		"install/helm/agones/values.yaml",
+		"sdks/nodejs/package.json",
+		"sdks/nodejs/package-lock.json",
+		"sdks/unity/package.json",
+		"sdks/csharp/sdk/AgonesSDK.nuspec",
+		"sdks/csharp/sdk/csharp-sdk.csproj",
+	}
+
+	for _, filename := range files {
+		var err error
+		if releaseType == "before" {
+			err = UpdateValueInFileBeforeRelease(filename, initialVersion)
+		} else if releaseType == "after" {
+			err = UpdateVersionInFileAfterRelease(filename, initialVersion)
+		} else {
+			log.Fatalf("Invalid release type. Please specify 'before' or 'after'.")
+		}
+
+		if err != nil {
+			log.Fatalf("Error updating file %s: %s\n", filename, err.Error())
+		}
+	}
+}
+
+func UpdateValueInFileBeforeRelease(filename string, valueToUpdate string) error {
```

Comment:
There is a lot of duplication in the `BeforeRelease` and `AfterRelease` variants - you should be able to factor these together. Consider that all this really is is "replace regexp with value" and what you need is a function that takes a regexp and replaces - the regexp and replacement value are the parameters. Then you can set those parameters based on whether it's before/after and whether it's JSON, which seems to be the only differences.

Reply:
I tried my best to avoid the duplication😀

Please let me know if there are any areas of the code that you think could be improved. Thanks!